samza-core/src/main/java/org/apache/samza/storage/blobstore/util/BlobStoreUtil.java

```diff
@@ -182,13 +182,13 @@ public CompletableFuture<SnapshotIndex> getSnapshotIndex(String blobId, Metadata
     return FutureUtil.executeAsyncWithRetries(opName, () -> {
       ByteArrayOutputStream indexBlobStream = new ByteArrayOutputStream(); // no need to close ByteArrayOutputStream
       return blobStoreManager.get(blobId, indexBlobStream, metadata, getDeleted).toCompletableFuture()
-          .thenApplyAsync(f -> snapshotIndexSerde.fromBytes(indexBlobStream.toByteArray()), executor)
-          .handle((snapshotIndex, ex) -> {
-            if (ex != null) {
+          .thenApplyAsync(f -> {
+            try {
+              return snapshotIndexSerde.fromBytes(indexBlobStream.toByteArray());
+            } catch (Exception ex) {
```

Comment:
Compared to previous change, seems this will only log the blobId if the serialization failed. Is that the case? Maybe we keep the previous exception handling, just updating the exception msg to be something like "Unable to get SnapshotIndex with blob ID...."?

Reply:
Fixed this. Only the exception message needed fixing.